ingestion/src/metadata/ingestion/source/deltalake.py

```diff
@@ -152,7 +197,59 @@ def _fetch_view_schema(self, view_name: str) -> Optional[Dict]:
                 view_detail[row_dict["col_name"]] = row_dict["data_type"]
             if "# Detailed Table" in row_dict["col_name"]:
                 col_details = True
-        return view_detail
+        return view_detail.get("View Text")
+
+    def _check_col_length(self, datatype, col_raw_type):
+        if datatype and datatype.upper() in {"CHAR", "VARCHAR", "BINARY", "VARBINARY"}:
+            try:
+                return col_raw_type.length if col_raw_type.length else 1
+            except AttributeError:
+                return 1
+
+    def _get_col_info(self, row):
+        parsed_string = ColumnTypeParser._parse_datatype_string(row["data_type"])
+        column = None
+        if parsed_string:
+            parsed_string["dataLength"] = self._check_col_length(
+                parsed_string["dataType"], row["data_type"]
+            )
+            if row["data_type"] == "array":
+                array_data_type_display = (
+                    repr(row["data_type"])
+                    .replace("(", "<")
+                    .replace(")", ">")
```

Comment:
instead of replacing these items directly here, it might be easier to follow to have a dictionary and use it everywhere. IDK if I've seen a similar recipe elsewhere 🤔 

even if it is just for this file, let's do something like:

```python
replace_array = {
    "(", "<",
     ...
}

...

for original, new in replace_array.items():
    data.replace(original, new)
```